kafka-adapter/kafka.go

```
@@ -0,0 +1,69 @@
+package main
+
+import (
+	//"fmt"
+	"encoding/json"
+	"github.com/Shopify/sarama"
+	"github.com/ngaut/log"
+	"strings"
+)
+
+const (
+	timeFormat = "2006-01-02 15:04:05"
+)
+
+//CreateKafkaProduce create a kafka produce
+func (r *Run) CreateKafkaProduce() error {
+	config := sarama.NewConfig()
+	config.Producer.RequiredAcks = sarama.WaitForLocal
+	config.Producer.Return.Successes = true
+	config.Producer.Partitioner = sarama.NewManualPartitioner
+	var err error
+	r.KafkaClient, err = sarama.NewSyncProducer(strings.Split(*kafkaAddress, ","), config)
+	return err
+}
+
+//PushKafkaMsg push message to kafka cluster
+func (r *Run) PushKafkaMsg(msg string) error {
+	kafkaMsg := &sarama.ProducerMessage{
+		Topic: *kafkaTopic,
+		Value: sarama.StringEncoder(msg),
+	}
+	log.Debugf("get kafka mssage %s", msg)
+	_, _, err := r.KafkaClient.SendMessage(kafkaMsg)
+	return err
+}
+
+//TransferData transfer alert to kafka string
```

If error, what happends

---

you can return an error in transferData